src/common/transformations/include/transformations/op_conversions/reduce_modification_for_false_keepdims.hpp

```diff
@@ -0,0 +1,143 @@
+public:
+    OPENVINO_RTTI("ModifyReduceMinForFalseKeepDims", "0");
+    ModifyReduceMinForFalseKeepDims();
+};
+
+class ov::pass::ModifyReduceLogicalAndForFalseKeepDims : public ModifyReduceBase {
+public:
+    OPENVINO_RTTI("ModifyReduceLogicalAndForFalseKeepDims", "0");
+    ModifyReduceLogicalAndForFalseKeepDims();
+};
+
+class ov::pass::ModifyReduceLogicalOrForFalseKeepDims : public ModifyReduceBase {
+public:
+    OPENVINO_RTTI("ModifyReduceLogicalOrForFalseKeepDims", "0");
+    ModifyReduceLogicalOrForFalseKeepDims();
+};
+
+class ov::pass::ModifyReduceForFalseKeepDims : public ov::pass::GraphRewrite {
+public:
+    OPENVINO_RTTI("ModifyReduceForFalseKeepDims", "0");
+    ModifyReduceForFalseKeepDims() {
+        add_matcher<ModifyReduceMeanForFalseKeepDims>();
+        add_matcher<ModifyReduceSumForFalseKeepDims>();
+        add_matcher<ModifyReduceMaxForFalseKeepDims>();
+        add_matcher<ModifyReduceMinForFalseKeepDims>();
+        add_matcher<ModifyReduceLogicalAndForFalseKeepDims>();
+        add_matcher<ModifyReduceLogicalOrForFalseKeepDims>();
+    }
+};
+
+template <class T>
+ov::matcher_pass_callback ModifyReduceBase::modify_reduce_for_false_keepdims() {
+    return [&](ov::pass::pattern::Matcher& m) {
+        auto reduce = std::dynamic_pointer_cast<T>(m.get_match_root());
+        if (!reduce)
+            return false;
+
+        auto input = reduce->input_value(0);
+        const auto input_shape = input.get_shape();
+        const auto reduce_shape = reduce->output(0).get_shape();
+        const auto input_rank = input.get_partial_shape().rank().get_length();
+
+        auto axes_vector = reduce->get_reduction_axes().to_vector();
+        std::sort(axes_vector.begin(), axes_vector.end());
+
+        if (is_unsupported_reordered_axes(axes_vector, input_rank, (input_rank - 2)) && input_shape.size() < 6) {
+            ngraph::NodeVector new_ops;
+
+            // Reduce
+            auto reduce_const =
+                ov::opset10::Constant::create(ov::element::i64, ov::Shape{axes_vector.size()}, axes_vector);
+            input = std::make_shared<T>(input, reduce_const, true);
+            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name());
+            new_ops.push_back(input.get_node_shared_ptr());
+
+            // Reshape
+            auto reshape_shape = ov::Shape(input_shape.size(), 1);
+            // Expected that a feature axis is only un-reduced.
+            reshape_shape[0] = reduce_shape[0];
+            input = std::make_shared<ov::opset10::Reshape>(
+                input,
+                ov::opset10::Constant::create(ov::element::i64, ov::Shape{reshape_shape.size()}, reshape_shape),
+                false);
+
+            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name() + "_reshape_false_keepdims");
+            new_ops.push_back(input.get_node_shared_ptr());
+
+            copy_runtime_info(reduce, new_ops);
+            reduce->output(0).replace(input);
```

Comment:
I think we need to return true here. isn't it?

Reply:
Sorry I missed it. It is added to PR for release/2022.2.
I might remove it by mistake. I will add it.